xray/src/generation.rs

```diff
@@ -127,7 +128,27 @@ pub trait ColoringStrategy: Send {
     // the pixel (x, y) in the final tile image.
     fn get_pixel_color(&self, x: u32, y: u32, background_color: Color<u8>) -> Color<u8>;
 
-    fn attributes(&self) -> Vec<&'static str>;
+    fn attributes(&self) -> HashSet<String>;
```

Comment:
Let's have a default implementation which returns an empty set so we don't have to do it twice downstairs?

Reply:
Done.